contracts/registry/src/registry.rs

```diff
@@ -400,10 +402,74 @@ impl SBTRegistry for Contract {
     /// Must be called by an SBT contract.
     /// Must emit `Revoke` event.
     /// Must also emit `Burn` event if the SBT tokens are burned (removed).
-    fn sbt_revoke(&mut self, token: Vec<TokenId>, burn: bool) {
+    fn sbt_revoke(&mut self, tokens: Vec<TokenId>, burn: bool) {
         let issuer = env::predecessor_account_id();
-        self.assert_issuer(&issuer);
-        env::panic_str("not implemented");
-        // add events
+        let issuer_id = self.assert_issuer(&issuer);
+        let mut tokens_burned = 0;
```

Comment:
```suggestion
        let tokens_burned = tokens.len();
```

Reply:
also, let's move this 1 line below (just before the for loop)

Reply:
I think we need to make sure how many tokens were revoked in case caller tries to revoke tokens that does not exist. That's why we cannot assume that `tokens_passed_to_method.len()` == `tokens_burned`

Reply:
then this line will fail:
```
let token_object = self.get_token(issuer_id, token);
```